xclim/analog.py

```diff
@@ -435,17 +447,31 @@ def kolmogorov_smirnov(x: np.ndarray, y: np.ndarray) -> float:
     :cite:cts:`fasano_multidimensional_1987`
     """
 
-    def pivot(x, y):
-        nx, d = x.shape
-        ny, d = y.shape
+    def pivot(_x: np.ndarray, _y: np.ndarray) -> float:
+        """Pivot function to compute the KS statistic.
```

Comment:
It's mostly to differentiate that the variables being used in this sub-function are not the same as those in the parent function. Also, this prevents situations where if someone were to remove the call signature of this function, it would become invalid (whereas previously, the function would have instead used the variables from the parent context).